internal/controller/policyserver_controller.go

```diff
@@ -176,53 +176,121 @@ func (r *PolicyServerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+		Watches(&policiesv1.ClusterAdmissionPolicyGroup{}, handler.EnqueueRequestsFromMapFunc(r.enqueueClusterAdmissionPolicyGroup)).
 		Complete(r)
 	if err != nil {
 		return errors.Join(errors.New("failed enrolling controller with manager"), err)
 	}
 	return nil
 }
 
+func (r *PolicyServerReconciler) enqueueAdmissionPolicy(_ context.Context, object client.Object) []reconcile.Request {
+	// The watch will trigger twice per object change; once with the old
+	// object, and once the new object. We need to be mindful when doing
+	// Updates since they will invalidate the newer versions of the
+	// object.
+	policy, ok := object.(*policiesv1.AdmissionPolicy)
+	if !ok {
+		r.Log.Info("object is not type of AdmissionPolicy: %+v", "policy", policy)
+		return []ctrl.Request{}
+	}
+
+	return []ctrl.Request{
+		{
+			NamespacedName: client.ObjectKey{
+				Name: policy.Spec.PolicyServer,
+			},
+		},
+	}
+}
+
+func (r *PolicyServerReconciler) enqueueAdmissionPolicyGroup(_ context.Context, object client.Object) []reconcile.Request {
+	// The watch will trigger twice per object change; once with the old
+	// object, and once the new object. We need to be mindful when doing
+	// Updates since they will invalidate the newer versions of the
+	// object.
+	policy, ok := object.(*policiesv1.AdmissionPolicyGroup)
+	if !ok {
+		r.Log.Info("object is not type of AdmissionPolicyGroup: %+v", "policy", policy)
+		return []ctrl.Request{}
+	}
+
+	return []ctrl.Request{
+		{
+			NamespacedName: client.ObjectKey{
+				Name: policy.Spec.PolicyServer,
+			},
+		},
+	}
+}
+func (r *PolicyServerReconciler) enqueueClusterAdmissionPolicy(_ context.Context, object client.Object) []reconcile.Request {
+	// The watch will trigger twice per object change; once with the old
+	// object, and once the new object. We need to be mindful when doing
+	// Updates since they will invalidate the newer versions of the
+	// object.
+	policy, ok := object.(*policiesv1.ClusterAdmissionPolicy)
+	if !ok {
+		r.Log.Info("object is not type of ClusterAdmissionPolicy: %+v", "policy", policy)
+		return []ctrl.Request{}
+	}
+
+	return []ctrl.Request{
+		{
+			NamespacedName: client.ObjectKey{
+				Name: policy.Spec.PolicyServer,
+			},
+		},
+	}
+}
+
+func (r *PolicyServerReconciler) enqueueClusterAdmissionPolicyGroup(_ context.Context, object client.Object) []reconcile.Request {
+	// The watch will trigger twice per object change; once with the old
+	// object, and once the new object. We need to be mindful when doing
+	// Updates since they will invalidate the newer versions of the
+	// object.
+	policy, ok := object.(*policiesv1.ClusterAdmissionPolicyGroup)
+	if !ok {
+		r.Log.Info("object is not type of ClusterAdmissionPolicyGroup: %+v", "policy", policy)
+		return []ctrl.Request{}
+	}
+
+	return []ctrl.Request{
+		{
+			NamespacedName: client.ObjectKey{
+				Name: policy.Spec.PolicyServer,
+			},
+		},
+	}
+}
+
 // getPolicies returns all admission policies and cluster admission
```

Comment:
quibble: should mention AdmissionPolicies, ClusterAdmissionPolicies, AdmissionPolicyGroups, ClusterAdmissionPolicGroups